sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -632,6 +634,29 @@ object ScalaJSPluginInternal {
     }
   }
 
+  protected def javaSysProps2JSLib(javaOptions: Seq[String]): VirtualJSFile = {
+    val javaSysPropsPattern = "-D([^=]*)=(.*)".r
+    val javaSysProps = javaOptions.map {
+      case javaSysPropsPattern(propName, propValue) => (propName, propValue)
+    }
+    val defaultSysProps = List(
+      "java.vm.version" -> ScalaJSVersions.current,
+      "os.name" -> java.lang.System.getProperty("os.name"),
+      "os.arch" -> java.lang.System.getProperty("os.arch"),
+      "os.version" -> java.lang.System.getProperty("os.version")
```

Comment:
`os` stuff should not be present, IMO. They would disappear if we run with something else than the sbt plugin. But there's no way we can set that from the tools, either. So I would just not support them.